src/NodaTime.Test/DurationTest.Construction.cs

```diff
@@ -6,16 +6,57 @@
 using NUnit.Framework;
 using System.Numerics;
 using static NodaTime.NodaConstants;
+using System.Linq;
 
 namespace NodaTime.Test
 {
     partial class DurationTest
     {
+        // Test cases for factory methods. In general, we want to check the limits, very small, very large
+        // and medium values, with a mixture of positive and negative, "on and off" day boundaries.
+
+        private static readonly int[] DayCases = {
+            Duration.MinDays, -3000 * 365, -100, -1, 0, 1, 100, 3000 * 365, Duration.MaxDays
+        };
+
+        private static readonly int[] HourCases = {
```

Comment:
Any reason we can't use `GenerateCases()` here? (and for that matter, for `DayCases` above?)

Oh, because these are `int[]`. Okay.